### ISSUE TITLE
quay.io public docker images broken? 

### ISSUE THREAD

**Issue by preston** [coreos/dex#301]
If you `docker pull quay.io/coreos/dex:latest` and then `docker run ...` the image, you'll get a nasty:

> Error response from daemon: No command specified

What gives? It doesn't look like the Dockerfile includes a CMD for the images on quay.io. I'm on Docker v1.9.1.

**Comment by ericchiang** [comment]
For deployments, dex is comprised of several Docker containers working together, each with different CMD fields. `quay.io/coreos/dex` is used as a base for these.

What are you trying to use the image for?

**Comment by preston** [comment]
Just trying to evaluate it at the moment. Is there a different repo with instructions on getting up and running with the correct set of images? Thanks much for your help!

**Comment by ericchiang** [comment]
The [getting started](https://github.com/coreos/dex/blob/master/Documentation/getting-started.md) doc is a good place to look. We currently don't have images available for this, though that might not be a bad idea in the future.

**Comment by preston** [comment]
Yeah I read through that but it doesn't really address Docker. Sounds like I'll need to go the manual way for now. It would indeed be super cool to provide Docker-specific page for folks like me that just want to deploy with a set of provided environment variables.

**Comment by ericchiang** [comment]
Yeah, it'd be pretty easy for us to provide containerized examples in the getting started guide for ease of use.

**Comment by jmheidly** [comment]
Hi guys,

I've been really struggling to dockerize and run dex components along with postgre. Any easy method?!

**Comment by ericchiang** [comment]
This is as far as I got.

```bash
# Run the Postgres container.
POSTGRES_CID=$( sudo docker run -d -P quay.io/coreos/postgres )
# Determine what port Postgres is running on.
POSTGRES_PORT=$( sudo docker inspect $POSTGRES_CID | jq '.[].NetworkSettings.Ports."5432/tcp"' | jq -r .[].HostPort )
# Determine what host to use (boot2docker/dockermachine vs native).
if [ "$DOCKER_HOST" != "" ]; then
    # Look for hostname of docker VM.
    DOCKER_HOSTNAME=$( echo -n ${DOCKER_HOST#tcp://} | cut -f1 -d":" )
else
    # Look for hostname of docker bridge.
    DOCKER_HOSTNAME=$( ip addr show docker0 | grep 'inet ' | awk '{ print $2 }' | cut -f1 -d"/" )
fi

# Create secrets
DEX_KEY_SECRET=$(dd if=/dev/random bs=1 count=32 2>/dev/null | base64 | tr -d '\n')
DEX_OVERLORD_ADMIN_API_SECRET=$(dd if=/dev/random bs=1 count=128 2>/dev/null | base64 | tr -d '\n')

DEX_DB_URL="postgres://postgres@$POSTGRES_HOST:$POSTGRES_PORT/postgres?sslmode=disable"

# Create a overlord.
sudo docker run -d -p 5557 quay.io/coreos/dex /opt/dex/bin/dex-overlord --db-url "$DEX_DB_URL" --key-secrets="$DEX_KEY_SECRET"

# Create a local connector.
echo '[{"type":"local","id":"local"}]' | sudo docker run --rm -i quay.io/coreos/dex /opt/dex/bin/dexctl --db-url="$DEX_DB_URL" set-connector-configs -

# Start a dex worker
sudo docker run --rm -i quay.io/coreos/dex /opt/dex/bin/dex-worker --db-url="$DEX_DB_URL" --key-secrets=$DEX_KEY_SECRET
```

Unfortunately the last command errors out with

```
$ sudo docker run --rm -i quay.io/coreos/dex /opt/dex/bin/dex-worker --db-url="$DEX_DB_URL" --key-secrets=$DEX_KEY_SECRET
WARN: Running with no limit on: database idle connections
WARN: Running with no limit on: database open connections
FATAL: Unable to build Server: open ./static/fixtures/emailer.json: no such file or directory
```

You'd have to mount the email.json file. Blah.

I've opened #388 to address this.

At the moment, I'd suggest just using a Go installation to compile dex and run postgres with docker/rkt. We're aware that dex's configuration is a pain point and have a few issues open to fix this.

**Comment by jmheidly** [comment]
Oh great. Thank you very much for the feedback.

I'll see how this setup goes and let you know.

**Comment by aarongreenlee** [comment]
I've been wanting to checkout dex since it was announced. This week, I am on vacation and thought I'd make some time to investigate. I ran into the same trouble as @preston in that I was hoping I could fire up a container and tap into it from my app for a quick evaluation.

**Comment by ericchiang** [comment]
@aarongreenlee sorry, yes we know this is a pain point. As noted above, dex is currently comprised of several different services: dexctl, dex-overlord, dex-worker, a database, and client applications.

This makes it hard to produce a good single container user experience, particularly since we develop for orchestrated environments. With that in mind there was a lot of work on the [dex on kubernetes documentation](https://github.com/coreos/dex/tree/master/contrib/k8s) contributed by @andrewstuart that you may want to look at.

For the Docker image on quay.io, perhaps we should provide a entrypoint that runs dex in development mode with an in memory DB and a pre-set list of clients? Since all of our internal applications override the entrypoint this wouldn't impact us. cc @sym3tri
